cdap-app-templates/cdap-etl/cdap-data-streams/src/main/java/co/cask/cdap/datastreams/SparkStreamingPipelineDriver.java

```diff
@@ -142,10 +180,24 @@ public JavaStreamingContext call() throws Exception {
         }
         if (checkpointDir != null) {
           jssc.checkpoint(checkpointDir);
+          jssc.sparkContext().hadoopConfiguration().set("fs.defaultFS", checkpointDir);
         }
         return jssc;
       }
     };
-    return checkpointDir == null ? contextFunction.call() : StreamingCompat.getOrCreate(checkpointDir, contextFunction);
+    return checkpointDir == null
+      ? contextFunction.call()
+      : StreamingCompat.getOrCreate(checkpointDir, contextFunction, context.hadoopConfiguration());
+  }
+
+  private static String appendToDirectory(String checkpointDirectory, String suffix) {
```

Comment:
Better use the `Path` object. `new Path(new Path(dir), suffix)`

Reply:
Thanks